src/metadata.rs

```diff
@@ -33,6 +54,27 @@ where
     })
 }
 
+/// Create the kubernetes recommended labels:
+/// - app.kubernetes.io/name - The name of the application e.g. mysql
+/// - app.kubernetes.io/instance - A unique name identifying the instance of an application e.g. mysql-abcxzy
+/// - app.kubernetes.io/version - The current version of the application (e.g., a semantic version, revision hash, etc.) e.g. 5.7.21
+/// - app.kubernetes.io/component - The component within the architecture e.g. database
+/// - app.kubernetes.io/part-of - The name of a higher level application this one is part of e.g. wordpress
+/// - app.kubernetes.io/managed-by - The tool being used to manage the operation of an application e.g. helm
+///
+fn get_recommended_labels<T>(resource: &T) -> OperatorResult<BTreeMap<String, String>>
```

Comment:
Could you move this to labels.rs (if you think it makes sense) and maybe remove the labels that are not set?